base/Process.php

```diff
@@ -99,19 +99,16 @@ public function isRunning(): bool {
   }
 
   public function stop(): void {
-    if (!$this->isRunning()) {
-      return;
-    }
-
     if (is_resource($this->stdin)) {
       pclose($this->stdin);
+      $this->stdin = null;
     }
     if (is_resource($this->stdout)) {
       pclose($this->stdout);
+      $this->stdout = null;
     }
-    $pid = $this->getPid();
-    if ($pid !== null) {
-      posix_kill($pid, SIGTERM);
+    if (is_resource($this->process)) {
+      proc_terminate($this->process);
```

Comment:
I don't think posix_kill is going to reap the subprocess. I could add a pcntl_wait after the kill, but I'm surprised proc_* isn't more portable.